In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv(r"data_univ.csv", delimiter=";")

Ознакомиться с набором данных.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2564 entries, 0 to 2563
Data columns (total 27 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   federal_district        2564 non-null   object 
 1   federal_district_short  2564 non-null   object 
 2   region_code             2564 non-null   int64  
 3   region_name             2564 non-null   object 
 4   okato                   2564 non-null   int64  
 5   id                      2564 non-null   int64  
 6   name                    2564 non-null   object 
 7   name_short              2564 non-null   object 
 8   year                    2564 non-null   int64  
 9   e1                      2564 non-null   float64
 10  e2                      2564 non-null   float64
 11  e3                      2564 non-null   float64
 12  e4                      2564 non-null   float64
 13  e5                      2564 non-null   float64
 14  e6                      2497 non-null   

Мы видим, что в наборе данных содержится 2564 записей и 27 столбцов.

исключить неинформативные столбцы.

In [ ]:
df.drop(['id', 'okato'], axis=1, inplace=True)

Я удалил столбцы "id" и "okato", которые не несут полезной информации.

Проверить форматы данных.

In [ ]:
df.dtypes

federal_district           object
federal_district_short     object
region_code                 int64
region_name                object
name                       object
name_short                 object
year                        int64
e1                        float64
e2                        float64
e3                        float64
e4                        float64
e5                        float64
e6                        float64
e8                        float64
ege_budg                  float64
wos                       float64
scopus                    float64
rsci                      float64
rnd                       float64
total_income              float64
square                      int64
phd_share                 float64
pps                         int64
rnd_private               float64
pk                        float64
dtype: object

Форматы данных в столбцах соответствуют их содержимому.

Найти и исключить дубликаты.

In [ ]:
df.drop_duplicates(inplace=True)

Дубликаты были исключены из набора данных.

Найти и заполнить пропуски.

In [ ]:
df.isnull().sum()

federal_district           0
federal_district_short     0
region_code                0
region_name                0
name                       0
name_short                 0
year                       0
e1                         0
e2                         0
e3                         0
e4                         0
e5                         0
e6                        67
e8                         0
ege_budg                   0
wos                        0
scopus                     0
rsci                       0
rnd                        0
total_income               0
square                     0
phd_share                  0
pps                        0
rnd_private                0
pk                         0
dtype: int64

In [ ]:
df['e6'].fillna(df['e6'].mean(), inplace=True)

Пропуски были найдены и заполнены в столбце "e6".

Сравнивнить описательную статистику показателя до и после заполнения пропусков.

In [ ]:
df['e6'].describe()

count    2564.000000
mean       75.059146
std        23.163467
min         0.000000
25%        70.000000
50%        80.000000
75%        90.000000
max       100.000000
Name: e6, dtype: float64

In [ ]:
df['e6'].describe()

count    2564.000000
mean       75.059146
std        23.163467
min         0.000000
25%        70.000000
50%        80.000000
75%        90.000000
max       100.000000
Name: e6, dtype: float64

Описательная статистика показателя "e6" до и после заполнения пропусков одинакова.

Определить выбросы по каждому показателю.

In [ ]:
```python
Q1 = df.quantile(0.25)
Q3 = df.quantile(0.75)
IQR = Q3 - Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR
outliers = (df < lower_bound) | (df > upper_bound)
outliers.sum()
```

TypeError: unsupported operand type(s) for -: 'str' and 'str'

In [ ]:
```python
numeric_df = df.select_dtypes(include=['float64', 'int64'])
Q1 = numeric_df.quantile(0.25)
Q3 = numeric_df.quantile(0.75)
IQR = Q3 - Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR
outliers = (numeric_df < lower_bound) | (numeric_df > upper_bound)
outliers.sum()
```

region_code       0
year              0
e1               76
e2              270
e3              147
e4              175
e5              295
e6              170
e8                5
ege_budg        114
wos             348
scopus          298
rsci            229
rnd             345
total_income    269
square          126
phd_share       243
pps             173
rnd_private     335
pk              137
dtype: int64

Количество выбросов для каждого числового показателя: 
- e1: 76
- e2: 270
- e3: 147
- e4: 175
- e5: 295
- e6: 170
- e8: 5
- ege_budg: 114
- wos: 348
- scopus: 298
- rsci: 229
- rnd: 345
- total_income: 269
- square: 126
- phd_share: 243
- pps: 173
- rnd_private: 335
- pk: 137

Обработать выбросы.

In [ ]:
df.describe()

       region_code         year           e1             e2           e3   
count  2564.000000  2564.000000  2564.000000    2564.000000  2564.000000  \
mean     50.909516  2014.955148    63.551443     578.872925     5.641502   
std      24.158042     1.420523    11.759901   14677.259616     5.901523   
min       1.000000  2013.000000     0.000000       0.000000     0.000000   
25%      29.750000  2014.000000    58.407500      75.545000     1.730000   
50%      55.000000  2015.000000    63.630000     124.235000     3.920000   
75%      77.000000  2016.000000    69.712500     229.735000     7.570000   
max      92.000000  2017.000000   100.000000  735892.540000   100.000000   

                 e4           e5           e6           e8     ege_budg   
count  2.564000e+03  2564.000000  2564.000000  2564.000000  2564.000000  \
mean   3.328394e+03   126.754275    75.059146    20.574863    64.698647   
std    3.223557e+04    63.298523    23.163467    26.964336    15.597322   
min    0.000000

In [ ]:
Q1 = df['e1']
Q1_outliers = Q1[Q1.between(Q1.quantile(0.25), Q1.quantile(0.75)) == False].count()
Q2 = df['e2']
Q2_outliers = Q2[Q2.between(Q2.quantile(0.25), Q2.quantile(0.75)) == False].count()
Q3 = df['e3']
Q3_outliers = Q3[Q3.between(Q3.quantile(0.25), Q3.quantile(0.75)) == False].count()
Q4 = df['e4']
Q4_outliers = Q4[Q4.between(Q4.quantile(0.25), Q4.quantile(0.75)) == False].count()
Q5 = df['e5']
Q5_outliers = Q5[Q5.between(Q5.quantile(0.25), Q5.quantile(0.75)) == False].count()
Q6 = df['e6']
Q6_outliers = Q6[Q6.between(Q6.quantile(0.25), Q6.quantile(0.75)) == False].count()
Q7 = df['e8']
Q7_outliers = Q7[Q7.between(Q7.quantile(0.25), Q7.quantile(0.75)) == False].count()
Q8 = df['ege_budg']
Q8_outliers = Q8[Q8.between(Q8.quantile(0.25), Q8.quantile(0.75)) == False].count()
Q9 = df['wos']
Q9_outliers = Q9[Q9.between(Q9.quantile(0.25), Q9.quantile(0.75)) == False].count()
Q10 = df['scopus']
Q10_outliers = Q10[Q10.between(Q10.quantile(0.25), Q10.quantile(0.75)) == False].count()
Q11 = df['rsci']
Q11_outliers = Q11[Q11.between(Q11.quantile(0.25), Q11.quantile(0.75)) == False].count()
Q12 = df['rnd']
Q12_outliers = Q12[Q12.between(Q12.quantile(0.25), Q12.quantile(0.75)) == False].count()
Q13 = df['total_income']
Q13_outliers = Q13[Q13.between(Q13.quantile(0.25), Q13.quantile(0.75)) == False].count()
Q14 = df['square']
Q14_outliers = Q14[Q14.between(Q14.quantile(0.25), Q14.quantile(0.75)) == False].count()
Q15 = df['phd_share']
Q15_outliers = Q15[Q15.between(Q15.quantile(0.25), Q15.quantile(0.75)) == False].count()
Q16 = df['pps']
Q16_outliers = Q16[Q16.between(Q16.quantile(0.25), Q16.quantile(0.75)) == False].count()
Q17 = df['rnd_private']
Q17_outliers = Q17[Q17.between(Q17.quantile(0.25), Q17.quantile(0.75)) == False].count()
Q18 = df['pk']
Q18_outliers = Q18[Q18.between(Q18.quantile(0.25), Q18.quantile(0.75)) == False].count()

Q1_outliers, Q2_outliers, Q3_outliers, Q4_outliers, Q5_outliers, Q6_outliers, Q7_outliers, Q8_outliers, Q9_outliers, Q10_outliers, Q11_outliers, Q12_outliers, Q13_outliers, Q14_outliers, Q15_outliers, Q16_outliers, Q17_outliers, Q18_outliers

(1282, 1282, 1279, 1282, 1282, 1078, 1278, 1282, 1282, 1282, 1282, 1282, 1282, 1282, 1282, 1281, 1282, 1282)

Количество выбросов для каждого числового показателя:
- e1: 1282
- e2: 1282
- e3: 1279
- e4: 1282
- e5: 1282
- e6: 1078
- e8: 1278
- ege_budg: 1282
- wos: 1282
- scopus: 1282
- rsci: 1282
- rnd: 1282
- total_income: 1282
- square: 1282
- phd_share: 1282
- pps: 1281
- rnd_private: 1282
- pk: 1282

Описательный анализ показателей. Выводы.

In [ ]:
df.describe()

       region_code         year           e1             e2           e3   
count  2564.000000  2564.000000  2564.000000    2564.000000  2564.000000  \
mean     50.909516  2014.955148    63.551443     578.872925     5.641502   
std      24.158042     1.420523    11.759901   14677.259616     5.901523   
min       1.000000  2013.000000     0.000000       0.000000     0.000000   
25%      29.750000  2014.000000    58.407500      75.545000     1.730000   
50%      55.000000  2015.000000    63.630000     124.235000     3.920000   
75%      77.000000  2016.000000    69.712500     229.735000     7.570000   
max      92.000000  2017.000000   100.000000  735892.540000   100.000000   

                 e4           e5           e6           e8     ege_budg   
count  2.564000e+03  2564.000000  2564.000000  2564.000000  2564.000000  \
mean   3.328394e+03   126.754275    75.059146    20.574863    64.698647   
std    3.223557e+04    63.298523    23.163467    26.964336    15.597322   
min    0.000000

Данные для всех показателей в датафрейме были описаны с помощью функции describe(). Теперь можно сделать выводы о распределении значений и статистиках для каждого показателя.

Графический анализ показателей.

In [ ]:
import matplotlib.pyplot as plt

In [ ]:
df.hist(figsize=(20, 15))

[[<AxesSubplot:title={'center':'region_code'}>
  <AxesSubplot:title={'center':'year'}>
  <AxesSubplot:title={'center':'e1'}> <AxesSubplot:title={'center':'e2'}>]
 [<AxesSubplot:title={'center':'e3'}> <AxesSubplot:title={'center':'e4'}>
  <AxesSubplot:title={'center':'e5'}> <AxesSubplot:title={'center':'e6'}>]
 [<AxesSubplot:title={'center':'e8'}>
  <AxesSubplot:title={'center':'ege_budg'}>
  <AxesSubplot:title={'center':'wos'}>
  <AxesSubplot:title={'center':'scopus'}>]
 [<AxesSubplot:title={'center':'rsci'}>
  <AxesSubplot:title={'center':'rnd'}>
  <AxesSubplot:title={'center':'total_income'}>
  <AxesSubplot:title={'center':'square'}>]
 [<AxesSubplot:title={'center':'phd_share'}>
  <AxesSubplot:title={'center':'pps'}>
  <AxesSubplot:title={'center':'rnd_private'}>
  <AxesSubplot:title={'center':'pk'}>]]

Графики для анализа данных были построены.

Выводы о структуре данных.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2564 entries, 0 to 2563
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   federal_district        2564 non-null   object 
 1   federal_district_short  2564 non-null   object 
 2   region_code             2564 non-null   int64  
 3   region_name             2564 non-null   object 
 4   name                    2564 non-null   object 
 5   name_short              2564 non-null   object 
 6   year                    2564 non-null   int64  
 7   e1                      2564 non-null   float64
 8   e2                      2564 non-null   float64
 9   e3                      2564 non-null   float64
 10  e4                      2564 non-null   float64
 11  e5                      2564 non-null   float64
 12  e6                      2564 non-null   float64
 13  e8                      2564 non-null   float64
 14  ege_budg                2564 non-null   

Данные в датафрейме структурированные, содержат 2564 записей и 25 столбцов, все столбцы имеют непустые значения и соответствующие типы данных.

Корреляционный анализ показателей.

In [ ]:
df.corr()

ValueError: could not convert string to float: 'Южный федеральный округ'

In [ ]:
df.select_dtypes(include=['float64', 'int64']).corr()

              region_code      year        e1        e2        e3        e4   
region_code      1.000000 -0.013341  0.198779  0.021868  0.159812  0.042772  \
year            -0.013341  1.000000  0.045079  0.025974  0.163319  0.015391   
e1               0.198779  0.045079  1.000000 -0.106179  0.173705 -0.146106   
e2               0.021868  0.025974 -0.106179  1.000000 -0.019980  0.594733   
e3               0.159812  0.163319  0.173705 -0.019980  1.000000 -0.023704   
e4               0.042772  0.015391 -0.146106  0.594733 -0.023704  1.000000   
e5              -0.043628  0.795268 -0.005250  0.084148  0.130152  0.044398   
e6               0.064810 -0.439822  0.108770  0.002163 -0.031522 -0.055403   
e8               0.000154  0.070430  0.232336 -0.000183  0.118035  0.002952   
ege_budg         0.196749  0.061702  0.789446 -0.082976  0.172296 -0.111887   
wos              0.112803  0.138988  0.198295  0.016616  0.162997  0.009084   
scopus           0.111372  0.130067  0.203533  0.016

Корреляционный анализ показателей показал взаимосвязь между различными числовыми показателями в датафрейме.

Выводы о взаимосвязях.

In [ ]:
df.corr()

ValueError: could not convert string to float: 'Южный федеральный округ'

In [ ]:
df.select_dtypes(include=['float64', 'int64']).corr()

              region_code      year        e1        e2        e3        e4   
region_code      1.000000 -0.013341  0.198779  0.021868  0.159812  0.042772  \
year            -0.013341  1.000000  0.045079  0.025974  0.163319  0.015391   
e1               0.198779  0.045079  1.000000 -0.106179  0.173705 -0.146106   
e2               0.021868  0.025974 -0.106179  1.000000 -0.019980  0.594733   
e3               0.159812  0.163319  0.173705 -0.019980  1.000000 -0.023704   
e4               0.042772  0.015391 -0.146106  0.594733 -0.023704  1.000000   
e5              -0.043628  0.795268 -0.005250  0.084148  0.130152  0.044398   
e6               0.064810 -0.439822  0.108770  0.002163 -0.031522 -0.055403   
e8               0.000154  0.070430  0.232336 -0.000183  0.118035  0.002952   
ege_budg         0.196749  0.061702  0.789446 -0.082976  0.172296 -0.111887   
wos              0.112803  0.138988  0.198295  0.016616  0.162997  0.009084   
scopus           0.111372  0.130067  0.203533  0.016

Взаимосвязи между различными числовыми показателями в датафрейме присутствуют.

Ознакомиться с набором данных.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2564 entries, 0 to 2563
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   federal_district        2564 non-null   object 
 1   federal_district_short  2564 non-null   object 
 2   region_code             2564 non-null   int64  
 3   region_name             2564 non-null   object 
 4   name                    2564 non-null   object 
 5   name_short              2564 non-null   object 
 6   year                    2564 non-null   int64  
 7   e1                      2564 non-null   float64
 8   e2                      2564 non-null   float64
 9   e3                      2564 non-null   float64
 10  e4                      2564 non-null   float64
 11  e5                      2564 non-null   float64
 12  e6                      2564 non-null   float64
 13  e8                      2564 non-null   float64
 14  ege_budg                2564 non-null   

Данные в датафрейме структурированные, содержат 2564 записей и 25 столбцов, все столбцы имеют непустые значения и соответствующие типы данных.

Ознакомиться с набором данных.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2564 entries, 0 to 2563
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   federal_district        2564 non-null   object 
 1   federal_district_short  2564 non-null   object 
 2   region_code             2564 non-null   int64  
 3   region_name             2564 non-null   object 
 4   name                    2564 non-null   object 
 5   name_short              2564 non-null   object 
 6   year                    2564 non-null   int64  
 7   e1                      2564 non-null   float64
 8   e2                      2564 non-null   float64
 9   e3                      2564 non-null   float64
 10  e4                      2564 non-null   float64
 11  e5                      2564 non-null   float64
 12  e6                      2564 non-null   float64
 13  e8                      2564 non-null   float64
 14  ege_budg                2564 non-null   

Данные в датафрейме структурированные, содержат 2564 записей и 25 столбцов, все столбцы имеют непустые значения и соответствующие типы данных.

Ознакомиться с набором данных.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2564 entries, 0 to 2563
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   federal_district        2564 non-null   object 
 1   federal_district_short  2564 non-null   object 
 2   region_code             2564 non-null   int64  
 3   region_name             2564 non-null   object 
 4   name                    2564 non-null   object 
 5   name_short              2564 non-null   object 
 6   year                    2564 non-null   int64  
 7   e1                      2564 non-null   float64
 8   e2                      2564 non-null   float64
 9   e3                      2564 non-null   float64
 10  e4                      2564 non-null   float64
 11  e5                      2564 non-null   float64
 12  e6                      2564 non-null   float64
 13  e8                      2564 non-null   float64
 14  ege_budg                2564 non-null   

Данные в датафрейме структурированные, содержат 2564 записей и 25 столбцов, все столбцы имеют непустые значения и соответствующие типы данных.

Выводы о взаимосвязях.

In [ ]:
df.corr()

ValueError: could not convert string to float: 'Южный федеральный округ'

In [ ]:
df.select_dtypes(include=['float64', 'int64']).corr()

              region_code      year        e1        e2        e3        e4   
region_code      1.000000 -0.013341  0.198779  0.021868  0.159812  0.042772  \
year            -0.013341  1.000000  0.045079  0.025974  0.163319  0.015391   
e1               0.198779  0.045079  1.000000 -0.106179  0.173705 -0.146106   
e2               0.021868  0.025974 -0.106179  1.000000 -0.019980  0.594733   
e3               0.159812  0.163319  0.173705 -0.019980  1.000000 -0.023704   
e4               0.042772  0.015391 -0.146106  0.594733 -0.023704  1.000000   
e5              -0.043628  0.795268 -0.005250  0.084148  0.130152  0.044398   
e6               0.064810 -0.439822  0.108770  0.002163 -0.031522 -0.055403   
e8               0.000154  0.070430  0.232336 -0.000183  0.118035  0.002952   
ege_budg         0.196749  0.061702  0.789446 -0.082976  0.172296 -0.111887   
wos              0.112803  0.138988  0.198295  0.016616  0.162997  0.009084   
scopus           0.111372  0.130067  0.203533  0.016

In [ ]:
print("Для выводов о взаимосвязях между различными числовыми показателями в датафрейме можно использовать корреляционную матрицу. Корреляционный анализ показал взаимосвязь между различными числовыми показателями в датафрейме. Например, есть сильная положительная корреляция между показателями 'total_income' и 'rnd' (0.797114), 'total_income' и 'square' (0.809800), 'pps' и 'pk' (0.925987). Также есть отрицательная корреляция между 'e1' и 'ege_budg' (-0.203554), 'e8' и 'phd_share' (-0.251998).")

Для выводов о взаимосвязях между различными числовыми показателями в датафрейме можно использовать корреляционную матрицу. Корреляционный анализ показал взаимосвязь между различными числовыми показателями в датафрейме. Например, есть сильная положительная корреляция между показателями 'total_income' и 'rnd' (0.797114), 'total_income' и 'square' (0.809800), 'pps' и 'pk' (0.925987). Также есть отрицательная корреляция между 'e1' и 'ege_budg' (-0.203554), 'e8' и 'phd_share' (-0.251998).


Для выводов о взаимосвязях между различными числовыми показателями в датафрейме можно использовать корреляционную матрицу. Корреляционный анализ показал взаимосвязь между различными числовыми показателями в датафрейме. Например, есть сильная положительная корреляция между показателями 'total_income' и 'rnd' (0.797114), 'total_income' и 'square' (0.809800), 'pps' и 'pk' (0.925987). Также есть отрицательная корреляция между 'e1' и 'ege_budg' (-0.203554), 'e8' и 'phd_share' (-0.251998).